cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Babbage/Transaction.hs

```diff
@@ -0,0 +1,110 @@
+import qualified Data.Text as Text
+import           System.FilePath ((</>))
+import qualified System.Info as SYS
+
+import           Hedgehog (Property)
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.File as H
+
+import qualified Data.Map as Map
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import           Testnet.Components.SPO
+import qualified Testnet.Process.Run as H
+import           Testnet.Process.Run
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+hprop_transaction :: Property
+hprop_transaction = H.integrationRetryWorkspace 0 "conway-stake-snapshot" $ \tempAbsBasePath' -> do
+  H.note_ SYS.os
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let
+    tempBaseAbsPath = makeTmpBaseAbsPath $ TmpAbsolutePath tempAbsPath'
+    era = BabbageEra
+    options = cardanoDefaultTestnetOptions
+      { cardanoNodes = cardanoDefaultTestnetNodeOptions
+      , cardanoSlotLength = 0.1
+      , cardanoNodeEra = AnyCardanoEra era -- TODO: We should only support the latest era and the upcoming era
+      }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    } <- cardanoTestnet options conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let sbe = ShelleyBasedEraBabbage
+
+  txbodyFp <- H.note $ work </> "tx.body"
+  txbodySignedFp <- H.note $ work </> "tx.body.signed"
+
+  void $ execCli' execConfig
+    [ "babbage", "query", "utxo"
+    , "--address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "utxo-1.json"
+    ]
+
+  utxo1Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-1.json"
+  UTxO utxo1 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo1Json
+  txin1 <- H.noteShow =<< H.headM (Map.keys utxo1)
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "build"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ head wallets
+    , "--tx-in", Text.unpack $ renderTxIn txin1
+    , "--tx-out", Text.unpack (paymentKeyInfoAddr (head wallets)) <> "+" <> show @Int 5_000_000
+    , "--out-file", txbodyFp
+    ]
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "sign"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-body-file", txbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--out-file", txbodySignedFp
+    ]
+
+  void $ execCli' execConfig
+    [ "babbage", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", txbodySignedFp
+    ]
+
+  H.threadDelay 1_000_000
```

Comment:
Given the short slot duration, consider asserting for created txo. Can use e.g. [getTxOutAtAddress](https://github.com/input-output-hk/antaeus/blob/828f950db4bd08062817125ecb984710539fe200/e2e-tests/test/Helpers/Query.hs#L179)

Reply:
Asserted the UTxO was created.

Reply:
Thanks. I think use of the cli will be less maintenance going forwards than the api directly :grin: